lib/contract.ts

```diff
@@ -1087,89 +1176,32 @@ export default class Contract {
 	 */
 	getNotSatisfiedChildRequirements(
 		contract: Contract,
-		options: { types: Set<string> } = { types: new Set() },
+		options: { types?: Set<string> } = {},
 	): any[] {
-		const conjuncts = reduce(
-			contract.getChildren(),
-			(accumulator, child) => {
-				return accumulator.concat(
-					child.metadata.requirements.compiled.getAll(),
-				);
-			},
-			contract.metadata.requirements.compiled.getAll(),
-		);
+		const conjuncts: Contract[] = contract.metadata.requirements.compiled
+			.getAll()
+			.concat(
+				contract
+					.getChildren()
+					.flatMap((child) => child.metadata.requirements.compiled.getAll()),
+			);
 		// (1) If the top level list of conjuncts is empty,
 		// then we can assume the requirements are fulfilled
 		// and stop without doing any further computations.
 		if (conjuncts.length === 0) {
 			return [];
 		}
-		// Utilities
-		const shouldEvaluateType = (type: string) =>
-			options.types ? options.types.has(type) : true;
 
 		const requirements: any[] = [];
-		/**
-		 * @summary Check if a matcher is satisfied
-		 * @function
-		 * @public
-		 *
-		 * @param {Object} matcher - matcher contract
-		 * @returns {Boolean} whether the matcher is satisfied
-		 *
-		 * @example
-		 * const matcher = Contract.createMatcher({
-		 *   type: 'sw.os',
-		 *   slug: 'debian'
-		 * })
-		 *
-		 * if (hasMatch(matcher)) {
-		 *   console.log('This matcher is satisfied!')
-		 * }
-		 */
-		const hasMatch = (matcher: Contract): boolean => {
-			// TODO: Write a function similar to findContracts
-			// that stops as soon as it finds one match
-			return (
-				this.findChildren(matcher).length > 0 ||
-				this.findChildrenWithCapabilities(matcher).length > 0
-			);
-		};
+
 		// (2) The requirements are specified as a list of objects,
 		// so lets iterate through those.
 		// This function uses a for loop instead of a more functional
 		// construct for performance reasons, given that we can freely
 		// break out of the loop as soon as possible.
 		for (const conjunct of conjuncts) {
```

Comment:
This could be done with a filter/map as well, I don't think performance would matter here so it might be ok but I'm also happy as-is but would suggest at least removing the `any[]` from `const requirements: any[] = [];` to improve the typings (it was avoiding that which prompted trying to return directly and let typescript do the inference)
```suggestion
		return conjuncts
			.filter((conjunct) => !this.isRequirementSatisfied(conjunct, options))
			.map((conjunct) => conjunct.raw.data)
```